### PR TITLE
feat: added named regex groups and substitute '_' for '.'

### PR DESCRIPTION
**Pull request by weliasz**
# Description

Added named regex groups and substitute '_' for '.'

## Type of change

Please delete options that are not relevant.

- [x] New feature

## How Has This Been Tested?

I changed the conf and verified proper parsing in Splunk

## Checklist

- [x] My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] I have run pre-commit on all files before creating the PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: splunk_connect_for_snmp_poller/manager/hec_sender.py [hunk 1]
```diff
@@ -218,28 +218,26 @@ def extract_additional_properties(fields, metric_name, metric_value, server_conf
...
-                fields["index_number"] = input_text
-                del fields["metric_name:" + metric_name]
-                fields["metric_name:" + stripped] = metric_value
+            entries = multi_key_lookup(
+                oid_families, (family, enricher_additional_varbinds)
+            )
+            if entries:
+                for entry in entries:
+                    if "regex" in entry:
+                        regex = entry["regex"]
```
Comment (omrozowicz-splunk) [inline splunk_connect_for_snmp_poller/manager/hec_sender.py]: I'm not sure if it's a good idea, but we can simplify that loop with:
```
regex_entries = [entry['regex'] for entry in entries if 'regex' in entry]
for regex in regex_entries:
    result = re.match(regex, input_text)
```

**Review by omrozowicz-splunk — APPROVED** [review]

File: splunk_connect_for_snmp_poller/manager/hec_sender.py [hunk 2]
```diff
@@ -215,31 +215,33 @@ def extract_additional_properties(fields, metric_name, metric_value, server_conf
...
-                        for index, item in enumerate(names_list):
-                            fields[item] = result.group(index + 1)
+                        for key, value in result.groupdict().items():
+                            fields[key] = value.replace("_", ".")
                         del fields["metric_name:" + metric_name]
                         fields["metric_name:" + stripped] = metric_value
-                        # TODO delete blow debug statement
-                        fields["old_metric_name:" + metric_name] = metric_value
                         continue
+            continue
```
Comment (omrozowicz-splunk) [inline splunk_connect_for_snmp_poller/manager/hec_sender.py]: Why do we need `continue` here?

Comment (weliasz) [reply splunk_connect_for_snmp_poller/manager/hec_sender.py]: Actually it should be break in both cases.

**Pull request merged**